### PR TITLE
support full alpha tcp connection

### PR DESCRIPTION
**Pull request by SafaSafari**
#### Description

add a filter to non-tcp connections for full alphabetics tcp connection
this depends of first word in tcp payload
after split by whitespace and checking length of that

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by SafaSafari** [comment]
In fact, this also applies to the SSH protocol.
However, the case I encountered was in reverse engineering an Android app that sent the information after Base64 encoding =)))

File: mitmproxy/addons/next_layer.py [hunk 1]
```diff
@@ -184,6 +184,8 @@ def s(*layers):
         probably_no_http = (
             # the first three bytes should be the HTTP verb, so A-Za-z is expected.
             len(data_client) < 3
+            # limit first word of http request len ex. OPTIONS, CONNECT (full alpha tcp connection)
+            or not len(data_client.split(b" ")[0]) <= 7
```
Comment (mhils) [inline mitmproxy/addons/next_layer.py:188]:
```suggestion
            # HTTP would require whitespace before the first newline
            # if we have neither whitespace nor a newline, it's also unlikely to be HTTP.
            or (data_client.find(b" ") >= data_client.find(b"\n"))
```

File: test/mitmproxy/addons/test_next_layer.py [hunk 2]
```diff
@@ -94,6 +94,8 @@
 
 dns_query = bytes.fromhex("002a01000001000000000000076578616d706c6503636f6d0000010001")
 
+tcp_fullalpha = b"AAAAAAAAAAAAAAAAAAAAAA=="
```
Comment (mhils) [inline test/mitmproxy/addons/test_next_layer.py:97]:
```suggestion
# Custom protocol with just base64-encoded messages
# https://github.com/mitmproxy/mitmproxy/pull/7087
custom_base64_proto = b"AAAAAAAAAAAAAAAAAAAAAA=="
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@
 
 ## Unreleased: mitmproxy next
 
+- Support full alpha tcp connection.
```
Comment (mhils) [inline CHANGELOG.md:10]:
```suggestion
- Tighten HTTP detection heuristic to better support custom TCP-based protocols.
```

**Review by mhils — COMMENTED** [review]
Thanks! SSH we already special-cased lately. But I do agree that we can improve here. :) 
len <= 7 isn't a great heuristic because HTTP methods can be longer (https://www.iana.org/assignments/http-methods/http-methods.xhtml). Please take a look at the suggestions below and let me know if that's a viable alternative!

**Comment by SafaSafari** [comment]
Yes, you're right. Decision-making based on fixed characteristics in HTTP requests is better.

**Review by mhils — APPROVED** [review]
Thanks! 🍰

**Pull request merged**